mmv1/third_party/terraform/fwprovider/framework_provider_billing_project_test.go.tmpl

```diff
@@ -0,0 +1,308 @@
+				),
+			},
+		},
+	})
+}
+
+func testAccFwProvider_billing_project_emptyStringValidation(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	billingProject := "my-billing-project-id"
+
+	// ensure all billing_project env vars set
+	t.Setenv("GOOGLE_BILLING_PROJECT", billingProject)
+
+	context := map[string]interface{}{
+		"billing_project": "", // empty string used
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccFwProvider_billing_project_inProviderBlock(context),
+				PlanOnly:    true,
+				ExpectError: regexp.MustCompile("expected a non-empty string"),
+			},
+		},
+	})
+}
+
+{{- if ne $.TargetVersionName "ga" }}
+func testAccFwProvider_billing_project_useWithAndWithoutUserProjectOverride(t *testing.T) {
+
+	randomString := acctest.RandString(t, 10)
+	contextUserProjectOverrideFalse := map[string]interface{}{
+		"org_id":                envvar.GetTestOrgFromEnv(t),
+		"billing_account":       envvar.GetTestBillingAccountFromEnv(t),
+		"user_project_override": "false", // Used in combo with billing_account
+		"random_suffix":         randomString,
+		// Firebase stuff below
+		"bundle_id":    "apple.app." + acctest.RandString(t, 5),
+		"display_name": "tf-test Display Name AppleAppConfig DataSource",
+		"app_store_id": 12345,
+		"team_id":      1234567890,
+	}
+
+	contextUserProjectOverrideTrue := map[string]interface{}{
+		"org_id":                envvar.GetTestOrgFromEnv(t),
+		"billing_account":       envvar.GetTestBillingAccountFromEnv(t),
+		"user_project_override": "true", // Used in combo with billing_account
+		"random_suffix":         randomString,
+		// Firebase stuff below
+		"bundle_id":    "apple.app." + acctest.RandString(t, 5),
+		"display_name": "tf-test Display Name AppleAppConfig DataSource",
+		"app_store_id": 12345,
+		"team_id":      1234567890,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t),
+		Steps: []resource.TestStep{
+			{
+				// Setup resources
+				// Neither user_project_override nor billing_project value used here
+				Config: testAccFwProvider_billing_project_useBillingProject_setup(contextUserProjectOverrideFalse),
+			},
+			{
+				// With user_project_override=true the Firebase data source CANNOT be used because quota is consumed
+				// from the newly provisioned project, and that project does not have the Firebase Management API enabled.
+				// The billing_project is used, leading to the error occurring, because user_project_override=true
+				Config:      testAccFwProvider_billing_project_useBillingProject_scenario(contextUserProjectOverrideTrue),
+				ExpectError: regexp.MustCompile("Error 403: Firebase Management API has not been used in project"),
+			},
```

Comment:
The failure [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11610#issuecomment-2386175893) appears to be because the text has a newline in the response from the API

Reply:
Should be resolved by [23efc9c](https://github.com/GoogleCloudPlatform/magic-modules/pull/11610/commits/23efc9cae1a5b52f7158ce031c88d901b0df3f91)